test/e2e/e2e_suite_test.go

```diff
@@ -58,3 +59,23 @@ func createTestCatalog(ctx context.Context, name string, imageRef string) (*cata
 	err := c.Create(ctx, catalog)
 	return catalog, err
 }
+
+func patchTestCatalog(ctx context.Context, name string, newImageRef string) (*catalogd.ClusterCatalog, error) {
+	// Fetch the existing ClusterCatalog
+	catalog := &catalogd.ClusterCatalog{}
+	err := c.Get(ctx, client.ObjectKey{Name: name}, catalog)
+	if err != nil {
+		return nil, err
+	}
+
+	// Update the ImageRef
+	catalog.Spec.Source.Image.Ref = newImageRef
+
+	// Patch the ClusterCatalog
+	err = c.Update(ctx, catalog)
+	if err != nil {
+		return nil, err
+	}
+
+	return catalog, err
+}
```

Comment:
I think this is a valid test case (where we patch the spec of a ClusterCatalog), but it does not cover the case I had in mind (and a case I think we need to cover).

What I have in mind here specifically is that the polling behavior alone is enough to trigger an upgrade of a ClusterExtension. Forcing a change by changing the image in the ClusterCatalog spec is a side-effect foils the true intention of the test.

This "polling triggers an upgrade" test is one of the most important features of OLM, so we unfortunately can't "fake" it my touching the spec.

Reply:
That was my first approach as well. Unfortunately, I can't find a way to change the `latest` image tag on a local registry. `crane` does not allow to change tags on local registries, hence I took this approach. Let me know if you you think this is still a valid test or we can approach this issue with a different solution. @joelanford

Reply:
> This "polling triggers an upgrade" test is one of the most important features of OLM, so we unfortunately can't "fake" it my touching the spec.

How is this accomplished in the current version of OLM? The use of make/sh/kinako(spelling), to get things into the cluster, the go to `run tests`, IMO blurs what is/isn't being tested, there aren't clear separation of duties in this model.

Correct me if I'm wrong, but I think you want an image loaded into the running cluster _after_ the test has executed? Is that correct? Or do you just want this test to run longer to see if the `polling` picks up the the image already in the registry?

Reply:
> How is this accomplished in the current version of OLM?

I'm not sure

> The use of make/sh/kinako(spelling), to get things into the cluster, the go to run tests, IMO blurs what is/isn't being tested, there aren't clear separation of duties in this model.

I agree that the number of moving parts is a problem. I'm fairly confident that if we wanted something a bit leaner, we could build all of this in Go.
1. Code in go that runs during an e2e that deploys an image registry in the cluster.
2. Client-go code that runs `kubectl port-forward` to give the e2e test a way to communicate with the registry.
3. go-containerregistry code to push images, move tags, etc.

> Correct me if I'm wrong, but I think you want an image loaded into the running cluster after the test has executed? Is that correct?

No, I don't think that is sufficient because the polling behavior of catalogd queries a remote registry. `kind load` would not make it visible to the polling process. What I'm after is $something that causes the digest of a tag to change mid way through the test. That could be one of:
1. Pre-populating the registry with two catalog images as part of the setup, and then during the test re-tagging the second image over the first.
2. Pushing the first image at `repo:tag` as part of the setup, and then pushing the second image at that same `repo:tag` during the test.